trieste/acquisition/function.py

```diff
@@ -130,6 +133,117 @@ def expected_improvement(model: ModelInterface, eta: tf.Tensor, at: QueryPoints)
     return (eta - mean) * normal.cdf(eta) + variance * normal.prob(eta)
 
 
+class MaxValueEntropySearch(SingleModelAcquisitionBuilder):
+    """
+    Builder for the max-value entropy search acqusiiton function (for function minimisation)
+    """
+
+    def __init__(self,search_space: SearchSpace, num_samples: int = 10, grid_size: int = 5000):
+        """
+        :param num_samples: integer determining how many samples to draw of the minimum 
+            (does not need to be large).
+        :param grid_size: number of random locations in grid used to fit the gumbel distribution 
+            and approximately generate the samples of the minimum (recommend scaling with problem dimension).
+        """
+        self._num_samples = num_samples
+        self._grid_size = grid_size
+        self._search_space = search_space
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ModelInterface
+    ) -> AcquisitionFunction:
+        """
+        Need to sample possible min-values from our posterior. To do this we implement a Gumbel sampler.
+        we approximate Pr(y*^hat<y) by Gumbel(alpha,beta) then sample from Gumbel.
+
+        :param dataset: Unused.
+        :param model: The model over the specified ``dataset``.
+        :return: The MES function.
+        """
+
+        # generate random grid to fit Gumbel distribution
+        # grid consists of already queried locations and grid_size other random locations
+        query_points = self._search_space.sample(self._grid_size)
+        query_points = tf.concat([dataset.query_points,query_points],0)
+        fmean, fvar = model.predict(query_points)
+        fsd = tf.math.sqrt(fvar)
+        idx = tf.math.argmax(-fmean[:tf.shape(dataset.query_points)[0]])
```

Comment:
i'd expect `fmean` to have the same length as `query_points`, and `argmax ~ - argmin`, so could this be
```suggestion
        idx = tf.argmin(fmean[:-1])
```

Reply:
also why are we not including the last row?

Reply:
This is just to find a good initialisation for the bisection search. We get the best location from the already queried points (not the whole sampled grid).

fmean has tf.shape(dataset.query_points)[0] + tf.shape(self_.grid_size)[0] points, so fmean[:-1] is not what we want here.

Reply:
yes, you're right about the shape, i missed that.